src/main/java/io/supertokens/webserver/WebserverAPI.java

```diff
@@ -481,7 +481,11 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
         }
         Logging.info(main, tenantIdentifier, "API ended: " + req.getRequestURI() + ". Method: " + req.getMethod(),
                 false);
-        RequestStats.getInstance(main, tenantIdentifier.toAppIdentifier()).updateRequestStats(true);
+        try {
+            RequestStats.getInstance(main, tenantIdentifier.toAppIdentifier()).updateRequestStats();
+        } catch (TenantOrAppNotFoundException e) {
+            throw new IllegalStateException(e); // should never happen
```

Comment:
remove this and just do nothing. And then remove the should never happen comment, and instead add a comment saying why we just ignore it